### DISCUSSION TITLE
Cache user badges to prevent disappearing issue

### DISCUSSION THREAD

**Discussion by jupjohn** [Chatterino/chatterino2#2522] [Ideas] [answered]
Occasionally user badges (sub, gift etc.) will not show in Chatterino when a user sends a message. I believe the reasoning behind this is Twitch not responding in time with the badges for that user in that channel. It would be nice to cache those badges for users instead of relying on Twitch every time a message is posted. Implementing a cache sounds like a good idea but could quickly become cumbersome due to large user/channel counts. 

Would love some discussion around this to hopefully reduce the impact of service events (such as yesterday) where badges aren't available.

**Comment by zneix** [comment]
I am against this - like for example, when I post a link in a chat, where only subs can post links, I sometimes get timed out because my sub badge disappears - showing a cached sub badge, but still being timed out would be ultra confusing.

**Reply by jupjohn** [reply]
Good point, didn't think of sub statuses being watched like that.

**Comment by pajlada** [comment, marked as answer]
I don't think there would be a simple and correct enough solution to warrant implementing this.

We would need some sort logic for expiring badges, in case a users sub actually expires, and any complexity that adds to the code would probably not be worth for this Twitch-specific issue.
